### PR TITLE
Split connector processing into planning and execution phases

### PR DESCRIPTION
**Pull request by mcusack-floe**
## Summary

This PR separates connector planning from table execution in the reconciler, enabling split control-plane and executor deployments while preserving the existing top-level jobs. It also strengthens durable queue behavior, execution routing, progress reporting, name normalization, and test coverage so planning, leasing, retries, cancellation, and remote execution behave more consistently and predictably.

## Type of Change

- [X] feat (new functionality)
- [ ] fix (bug fix)
- [ ] docs (documentation only)
- [X] refactor (no behavior change)
- [X] test (adds/updates tests)
- [ ] chore (build/tooling/maintenance)

## Testing

Describe how this was validated.

- [X] `make fmt`
- [X] `make verify`
- [X] Added/updated tests for changed behavior

## Deployment and Compatibility

- [X] No breaking changes
- [ ] Breaking changes (describe below)
- [ ] Config/env var changes (describe below)

## Checklist

- [X] PR is scoped and focused
- [X] Commit messages follow conventional commit style where practical
- [X] Documentation updated where needed
- [X] No credentials, tokens, or secrets are included
- [X] This PR does not disclose a security vulnerability publicly (see `SECURITY.md`)

### CODE REVIEW

File: core/connectors/common/src/main/java/ai/floedb/floecat/connector/common/ConnectorPlanningSupport.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  public static List<FloecatConnector.PlannedTableTask> planTableTasks(
+      FloecatConnector.TablePlanningRequest request, Function<String, List<String>> listTables) {
+    if (request == null) {
+      return List.of();
+    }
+
+    String sourceNamespaceFq = requireNonBlank(request.sourceNamespaceFq(), "source namespace");
+    String destinationNamespaceFq = blankToNull(request.destinationNamespaceFq());
+    if (!matchesNamespaceFilter(destinationNamespaceFq, request.destinationNamespacePaths())) {
```
Comment (mrouvroy-floe) [inline core/connectors/common/src/main/java/ai/floedb/floecat/connector/common/ConnectorPlanningSupport.java:35]: `planTableTasks` throws on destination namespace mismatch. 
For scoped reconciles this should usually resolve to “no matching tasks”, not a hard planner failure. 
Throwing here regresses prior scope behavior and turns a filter mismatch into an operational error.

File: reconciler/src/main/java/ai/floedb/floecat/reconciler/jobs/ReconcileJobStore.java [hunk 2]
```diff
@@ -260,34 +409,51 @@ public LeaseRequest(
...
-    public boolean matches(ReconcileExecutionPolicy policy, String pinnedExecutorId) {
+    public boolean matches(
+        ReconcileExecutionPolicy policy, String pinnedExecutorId, ReconcileJobKind jobKind) {
       ReconcileExecutionPolicy effective =
           policy == null ? ReconcileExecutionPolicy.defaults() : policy;
+      ReconcileJobKind effectiveJobKind =
+          jobKind == null ? ReconcileJobKind.PLAN_CONNECTOR : jobKind;
       boolean classMatches =
           executionClasses.isEmpty() || executionClasses.contains(effective.executionClass());
       boolean laneMatches = lanes.isEmpty() || lanes.contains(effective.lane());
```
Comment (mrouvroy-floe) [inline reconciler/src/main/java/ai/floedb/floecat/reconciler/jobs/ReconcileJobStore.java]: Lane matching is exact membership (lanes.contains(effective.lane())), but planner executor semantics are “supports any lane”. 
This can make PLAN_CONNECTOR jobs unleasable when floecat.reconciler.auto.execution-lane is set. 
We should model wildcard lane semantics explicitly.

File: reconciler/src/main/java/ai/floedb/floecat/reconciler/impl/ConnectorPlanningReconcileExecutor.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    return 10;
+  }
+
+  @Override
+  public Set<ReconcileJobKind> supportedJobKinds() {
+    return EnumSet.of(ReconcileJobKind.PLAN_CONNECTOR);
+  }
+
+  @Override
+  public Set<String> supportedLanes() {
```
Comment (mrouvroy-floe) [inline reconciler/src/main/java/ai/floedb/floecat/reconciler/impl/ConnectorPlanningReconcileExecutor.java:81]: supportedLanes() returns empty while supportsLane() returns true (any lane). With current lease matching this is inconsistent and can starve planner jobs. We need to align lease request + matching semantics so planner jobs remain leaseable under configured lanes.

File: reconciler/src/main/java/ai/floedb/floecat/reconciler/impl/ReconcilerScheduler.java [hunk 4]
```diff
@@ -187,6 +191,23 @@ private void submitLease(ReconcileJobStore.LeasedJob lease) {
   }
 
   void runLease(ReconcileJobStore.LeasedJob lease) {
+    ReconcileExecutor executor =
+        executorRegistry
+            .executorFor(lease)
+            .or(
+                () ->
+                    executorRegistry.orderedExecutors().stream()
+                        .filter(
```
Comment (mrouvroy-floe) [inline reconciler/src/main/java/ai/floedb/floecat/reconciler/impl/ReconcilerScheduler.java:205]: Fallback executor selection should also filter by `supportsJobKind(lease.jobKind)`.
Right now fallback checks class/lane/supports(lease), but job-kind compatibility should be explicit here too to avoid accidentally selecting an executor that cannot run the leased job type.

File: reconciler/src/main/java/ai/floedb/floecat/reconciler/impl/ConnectorPlanningReconcileExecutor.java [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+            .progressListener()
+            .onProgress(
+                planned,
+                0,
+                0,
+                0,
+                0,
+                "Planned table " + task.sourceNamespace() + "." + task.sourceTable());
+      }
+      return ExecutionResult.success(planned, 0, 0, 0, 0, "Planned " + planned + " table jobs");
```
Comment (mrouvroy-floe) [inline reconciler/src/main/java/ai/floedb/floecat/reconciler/impl/ConnectorPlanningReconcileExecutor.java]: Planner returns success for planned == 0. 
For scoped reconciles with a destination table filter that matches nothing, previous behavior surfaced a clear error (“No tables matched scope”). 
Let's restore explicit failure/diagnostic semantics for empty scoped plans.

File: service/src/main/java/ai/floedb/floecat/service/reconciler/impl/ReconcileControlImpl.java [hunk 6]
```diff
@@ -166,15 +172,28 @@ public Uni<StartCaptureResponse> startCapture(StartCaptureRequest request) {
...
+                    connectorRepo
+                        .getById(connectorId)
+                        .orElseThrow(
+                            () ->
+                                GrpcErrors.notFound(
+                                    correlationId,
+                                    GeneratedErrorMessages.MessageKey.CONNECTOR,
+                                    Map.of("id", connectorId.getId())));
+
                     var jobId =
```
Comment (mrouvroy-floe) [inline service/src/main/java/ai/floedb/floecat/service/reconciler/impl/ReconcileControlImpl.java:192]: StartCapture always enqueues PLAN_CONNECTOR but does not validate executor coverage for that job kind. If planner executor is disabled/missing, jobs can queue forever. 
We should fail fast (startup and/or enqueue-time) when no enabled executor can run PLAN_CONNECTOR.

File: service/src/main/java/ai/floedb/floecat/service/reconciler/impl/ReconcileControlImpl.java [hunk 7]
```diff
@@ -710,6 +761,135 @@ private static String fromProtoState(JobState state) {
...
+
+  private List<ReconcileJobStore.ReconcileJob> childJobsFor(
+      String accountId, ReconcileJobStore.ReconcileJob planJob) {
+    if (planJob == null || planJob.jobKind != ReconcileJobKind.PLAN_CONNECTOR) {
+      return List.of();
+    }
+    List<ReconcileJobStore.ReconcileJob> out = new ArrayList<>();
+    String nextToken = "";
+    do {
+      var page = jobs.list(accountId, 200, nextToken, planJob.connectorId, Set.of());
```
Comment (mrouvroy-floe) [inline service/src/main/java/ai/floedb/floecat/service/reconciler/impl/ReconcileControlImpl.java]: We scan child jobs via paginated jobs.list(...) for every plan job read. 
This can get expensive in listReconcileJobs and getReconcileJob at scale. 
Maybe we can add a store-side parent index / direct parent_job_id lookup to avoid repeated connector-wide scans ?

File: docs/reconciler.md [hunk 8]
```diff
@@ -83,23 +101,28 @@ Internally, the scheduler exposes `pollEvery` via `@Scheduled` (default every se
 
 ## Data Flow & Lifecycle
```
Comment (mrouvroy-floe) [inline docs/reconciler.md:114]: The flow should explicitly call out split-phase view behavior: connector-scoped view reconciliation runs once per top-level plan job, not per table child. Without this, contributors may reintroduce the prior regression by moving/skipping view sync in split mode.

File: docs/operations.md [hunk 9]
```diff
@@ -80,6 +83,10 @@ Recommended split deployment:
 - Executor plane: `QUARKUS_PROFILE=reconciler-executor`
 - Shared settings: same blob/kv backend, same reconcile auth token, executor nodes pointed at the control-plane gRPC host/port
 
+In the split model, the control plane owns top-level `PLAN_CONNECTOR` jobs and public reconcile
```
Comment (mrouvroy-floe) [inline docs/operations.md:86]: Please document the operational guardrail: if PLAN_CONNECTOR jobs are enqueued, at least one enabled executor must support that kind, and lane semantics must keep planner jobs leaseable. This is currently a hidden misconfiguration mode that manifests as stuck queued jobs.

File: service/src/test/java/ai/floedb/floecat/service/reconciler/jobs/ReconcilePlannerSchedulerTest.java
Comment (mrouvroy-floe) [inline service/src/test/java/ai/floedb/floecat/service/reconciler/jobs/ReconcilePlannerSchedulerTest.java:1]: - Missing test coverage for lane-edge behavior in split mode. 
Let's add a test proving planner jobs remain leaseable with non-empty configured execution lane (or fail-fast with explicit error if intentionally unsupported).
- Also worth  adding startup/enqueue-time test proving PLAN_CONNECTOR enqueue is rejected when no enabled executor supports that kind.

File: service/src/test/java/ai/floedb/floecat/service/reconciler/jobs/ReconcilePlannerSchedulerTest.java [hunk 10]
```diff
@@ -241,6 +249,41 @@ void runPlannerPassSkipsConnectorsWithDisabledPolicy() {
     assertEquals(List.of("conn-enabled"), enqueued);
   }
 
+  @Test
```
Comment (mrouvroy-floe) [inline service/src/test/java/ai/floedb/floecat/service/reconciler/jobs/ReconcilePlannerSchedulerTest.java:264]: This verifies execution policy propagation but not leaseability/runtime outcome. We need a companion test that exercises configured lane + planner executor support end-to-end (enqueue → lease), to prevent regressions where jobs silently remain queued.

File: service/src/test/java/ai/floedb/floecat/service/it/ConnectorIT.java
Comment (mrouvroy-floe) [inline service/src/test/java/ai/floedb/floecat/service/it/ConnectorIT.java:1]: We should: 
- Add an IT for scoped reconcile miss semantics in split mode: destination table filter matching no table should surface explicit failure diagnostics (not success/no-op). Current helper logic aggregates plan/children but doesn’t lock this contract.
- Add explicit IT for split-phase view-capable connector: top-level reconcile still syncs views once per plan job.

File: reconciler/src/main/java/ai/floedb/floecat/reconciler/jobs/impl/InMemoryReconcileJobStore.java
Comment (mrouvroy-floe) [inline reconciler/src/main/java/ai/floedb/floecat/reconciler/jobs/impl/InMemoryReconcileJobStore.java:407]: In-memory store drops view task context on failure transitions

In `markFailed`, the `ReconcileJob` constructor overload used does not include `viewTask`, so failed `EXEC_VIEW` jobs lose their source/destination view routing metadata (`viewTask` becomes empty). This regresses diagnostics/API visibility for failed view executions in the default in-memory store path.

File: service/src/main/java/ai/floedb/floecat/service/reconciler/impl/ReconcileExecutorControlImpl.java [hunk 11]
```diff
@@ -271,6 +301,40 @@ private static LeasedReconcileJob toProtoLease(ReconcileJobStore.LeasedJob lease
...
+    return ai.floedb.floecat.reconciler.rpc.ReconcileTableTask.newBuilder()
+        .setSourceNamespace(effective.sourceNamespace())
+        .setSourceTable(effective.sourceTable())
+        .setDestinationTableDisplayName(effective.destinationTableDisplayName())
+        .build();
+  }
+
+  private static ai.floedb.floecat.reconciler.rpc.ReconcileViewTask toProtoViewTask(
+      ReconcileViewTask viewTask) {
+    ReconcileViewTask effective = viewTask == null ? ReconcileViewTask.empty() : viewTask;
```
Comment (mrouvroy-floe) [inline service/src/main/java/ai/floedb/floecat/service/reconciler/impl/ReconcileExecutorControlImpl.java:332]: Parent cancel fails after planner job reaches JS_SUCCEEDED

`CancelReconcileJob` delegates to `jobs.cancel(...)` and returns NOT_FOUND when that call is empty. For plan jobs, scheduler marks the parent job succeeded immediately after enqueueing children, while aggregation still reports RUNNING/QUEUED via child states. At that point `jobs.cancel` treats the parent as terminal and returns empty, so users can no longer cancel a still-active top-level reconcile by its original job id.

File: reconciler/src/main/java/ai/floedb/floecat/reconciler/jobs/impl/InMemoryReconcileJobStore.java
Comment (mcusack-floe) [reply reconciler/src/main/java/ai/floedb/floecat/reconciler/jobs/impl/InMemoryReconcileJobStore.java:407]: Fixed. markFailed now preserves viewTask for failed EXEC_VIEW jobs, so routing/diagnostic metadata is retained in the in-memory store as well. Added coverage in reconciler/src/test/java/ai/floedb/floecat/reconciler/jobs/impl/InMemoryReconcileJobStoreTest.java.

File: service/src/main/java/ai/floedb/floecat/service/reconciler/impl/ReconcileExecutorControlImpl.java (hunk 11 above)
Comment (mcusack-floe) [reply service/src/main/java/ai/floedb/floecat/service/reconciler/impl/ReconcileExecutorControlImpl.java:332]: Fixed. Cancel-by-parent-id now still works after the planner job reaches JS_SUCCEEDED if child jobs are still active. The control path falls back to canceling active children instead of returning NOT_FOUND. Added coverage in service/src/test/java/ai/floedb/floecat/service/reconciler/impl/ReconcileControlImplTest.java.

**Review by mrouvroy-floe — APPROVED** [review]
